flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java

```diff
@@ -75,32 +92,584 @@ public void clean() {
+    String sqlBetween = String.format("SELECT * FROM %s WHERE id BETWEEN 1 AND 2 ", TABLE_NAME);
+    String explainBetween = getTableEnv().explainSql(sqlBetween);
+    String expectedFilter = "ref(name=\"id\") >= 1,ref(name=\"id\") <= 2";
+    Assert.assertTrue("Explain should contain the push down filter", explainBetween.contains(expectedFilter));
+
+    List<Object[]> resultBetween = sql(sqlBetween);
+    Assert.assertEquals("Should have 2 record", 2, resultBetween.size());
+
+    List<Object[]> expectedBetween = Lists.newArrayList();
+    expectedBetween.add(new Object[] {1, "iceberg", 10.0});
+    expectedBetween.add(new Object[] {2, "b", 20.0});
+    Assert.assertArrayEquals("Should produce the expected record", expectedBetween.toArray(), resultBetween.toArray());
+
+    Assert.assertEquals("Should create only one scan", 1, scanEventCount);
+    String expected = "(ref(name=\"id\") >= 1 and ref(name=\"id\") <= 2)";
+    Assert.assertEquals("Should contain the push down filter", expected, lastScanEvent.filter().toString());
+  }
+
+  @Test
+  public void testFilterPushDownNotBetween() {
+    String sqlNotBetween = String.format("SELECT * FROM %s WHERE id  NOT BETWEEN 2 AND 3 ", TABLE_NAME);
+    String explainNotBetween = getTableEnv().explainSql(sqlNotBetween);
+    String expectedFilter = "(ref(name=\"id\") < 2 or ref(name=\"id\") > 3)";
+    Assert.assertTrue("Explain should contain the push down filter", explainNotBetween.contains(expectedFilter));
+
+    List<Object[]> resultNotBetween = sql(sqlNotBetween);
+    Assert.assertEquals("Should have 1 record", 1, resultNotBetween.size());
+    Assert.assertArrayEquals("the not between Should produce the expected record", resultNotBetween.get(0),
+        new Object[] {1, "iceberg", 10.0});
+
+    Assert.assertEquals("Should create only one scan", 1, scanEventCount);
+    Assert.assertEquals("Should contain the push down filter", expectedFilter, lastScanEvent.filter().toString());
+  }
+
+  @Test
+  public void testFilterPushDownLike() {
+    String sqlLike = "SELECT * FROM " + TABLE_NAME + " WHERE data LIKE 'ice%' ";
+    String explainLike = getTableEnv().explainSql(sqlLike);
+    String expectedFilter = "ref(name=\"data\") startsWith \"\"ice\"\"";
+    Assert.assertTrue("the like sql Explain should contain the push down filter", explainLike.contains(expectedFilter));
+
+    sqlLike = "SELECT * FROM " + TABLE_NAME + " WHERE data LIKE 'ice%%' ";
+    List<Object[]> resultLike = sql(sqlLike);
+    Assert.assertEquals("Should have 1 record", 1, resultLike.size());
+    Assert.assertArrayEquals("The like result should produce the expected record",
+        new Object[] {1, "iceberg", 10.0}, resultLike.get(0));
+    Assert.assertEquals("Should create only one scan", 1, scanEventCount);
+    Assert.assertEquals("Should contain the push down filter", expectedFilter, lastScanEvent.filter().toString());
+  }
+
+  @Test
+  public void testFilterNotPushDownLike() {
```

Comment:
I think this is missing a case for `i%g`. It has a case for `%ice%g`, but I think it should test both.

Reply:
I add the test case